samples/dnn/seemore_superres.cpp

```diff
@@ -0,0 +1,202 @@
+using namespace std;
+
+const int WINDOW_OFFSET_X = 50;
+const int WINDOW_OFFSET_Y = 50;
+const int WINDOW_SPACING = 50;
+
+const string param_keys =
+    "{ help h          |                   | Print help message }"
+    "{ @alias          | seemoredetails    | Model alias from models.yml }"
+    "{ zoo             | ../dnn/models.yml | Path to models.yml file }"
+    "{ input i         |  chicky_512.png   | Path to input image }"
+    "{ model           |                   | Path to model file }";
+
+const string backend_keys = format(
+    "{ backend          | default | Choose one of computation backends: "
+    "default: automatically (by default), "
+    "openvino: Intel's Deep Learning Inference Engine (https://software.intel.com/openvino-toolkit), "
+    "opencv: OpenCV implementation, "
+    "vkcom: VKCOM, "
+    "cuda: CUDA, "
+    "webnn: WebNN }");
+
+const string target_keys = format(
+    "{ target           | cpu | Choose one of target computation devices: "
+    "cpu: CPU target (by default), "
+    "opencl: OpenCL, "
+    "opencl_fp16: OpenCL fp16 (half-float precision), "
+    "vpu: VPU, "
+    "vulkan: Vulkan, "
+    "cuda: CUDA, "
+    "cuda_fp16: CUDA fp16 (half-float preprocess) }");
+
+static Mat postprocessOutput(const Mat &output, const Size &originalSize)
+{
+    Mat squeezed;
+    if (output.dims == 4 && output.size[0] == 1)
+    {
+        vector<int> newShape = {output.size[1], output.size[2], output.size[3]};
+        squeezed = output.reshape(0, newShape);
+    }
+    else
+    {
+        squeezed = output.clone();
+    }
+
+    Mat outputImage;
+    vector<Mat> channels(3);
+    for (int i = 0; i < 3; i++)
+    {
+        channels[i] = Mat(squeezed.size[1], squeezed.size[2], CV_32F,
+                        squeezed.ptr<float>(i));
+    }
+    merge(channels, outputImage);
+
+    cvtColor(outputImage, outputImage, COLOR_RGB2BGR);
+
+    outputImage = max(0.0, min(1.0, outputImage));
+    outputImage.convertTo(outputImage, CV_8UC3, 255.0);
+
+    Size targetSize(originalSize.width * 4, originalSize.height * 4);
+    Mat result;
+    resize(outputImage, result, targetSize);
+
+    return result;
+}
+
+static Mat applySuperResolution(Net &net, const Mat &image, float scale, const Scalar &mean, bool swapRB, int width, int height)
+{
+    Mat blob = blobFromImage(image, scale, Size(width, height), mean, swapRB, false, CV_32F);
+
+    net.setInput(blob);
+    Mat output;
+    net.forward(output);
+
+    return postprocessOutput(output, Size(image.cols, image.rows));
+}
+
+static double calculateFontScale(const Mat &image)
+{
+    double baseScale = min(image.cols, image.rows) / 800.0;
+    return max(0.5, baseScale);
+}
+
+static void processFrame(Net &net, const Mat &frame, float scale, const Scalar &mean, bool swapRB, int width, int height)
+{
+    Mat result = applySuperResolution(net, frame, scale, mean, swapRB, width, height);
+    Mat frameWithText = frame.clone();
+    Mat resultWithText = result.clone();
```

Comment:
The frame and results are not used in the following code. No need to make a copy.